src/GUI.java

```diff
@@ -0,0 +1,133 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.*;
+import java.util.Objects;
+import javax.swing.border.*;
+
+public class GUI implements ActionListener {
+    JFrame frame;
+    JPanel panel;
+    JLabel labelInput, labelOutput;
+    JTextField funcOutput, funcInput;
+    JButton buttonCalc, buttonClear;
+    Font font = new Font("Open Sans Pro", Font.ITALIC, 20);
+    Font labelFont = new Font("Open Sans Pro", Font.ITALIC, 15);
+
+    //Konstruktor
+    public GUI(){
+        frame = new JFrame("CalculusMinds                     Derivation and Integration Calculator");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setSize(550, 420);
+        frame.setResizable(false);
+        frame.setLayout(null); //null: Position und Größe der Komponenten vollständig dem Entwickler überlassen
+
+        //labels für textfelder
+        labelInput = new JLabel("Enter what you would like to calculate");
+        labelInput.setBounds(50, 20, 430, 20);
+        labelInput.setForeground(Color.WHITE);
+        labelInput.setFont(labelFont);
+
+        labelOutput = new JLabel("Here is your solution");
+        labelOutput.setBounds(50, 250, 430, 20);
+        labelOutput.setForeground(Color.WHITE);
+        labelOutput.setFont(labelFont);
+
+        //Textfelder
+        funcInput = createCustomTextfield(50, 40, 430, 85);
+        funcOutput = createCustomTextfield(50, 270, 430, 85);
+        funcOutput.setEditable(false);
+
+        //Panel mit Buttons
+        panel = new JPanel();
+        panel.setBounds(50, 130, 430, 125);
+        panel.setBorder(BorderFactory.createEmptyBorder(5, 0, 5, 0));
+        panel.setBackground(Color.DARK_GRAY);
+
+        //Buttons
```

Comment:
Lieber sowas:

 try {
            buttonCalc = createCustomButton("Calculate");
            buttonClear = createCustomButton("Clear");
        } catch (OperationNotSupportedException e) {
            throw new RuntimeException("Failed to create button.");
        }